src/id.rs

```diff
@@ -111,6 +112,14 @@ impl Debug for PublicId {
     }
 }
 
+impl Deserialize for PublicId {
+    fn deserialize<D: Deserializer>(deserialiser: D) -> Result<Self, D::Error> {
+        let (public_encrypt_key, public_sign_key): (box_::PublicKey, sign::PublicKey) =
+            Deserialize::deserialize(deserialiser)?;
```

Comment:
Not that I know of; I'm quite uncomfortable with this too.  I opted for this approach since we're doing something similar in `ImmutableData` and `PrivAppendedData`, although those don't depend on the behaviour of handling tuples.  Instead they depend on the encoded format not adding anything to the member fields.

I created [a repo](https://gitlab.com/Fraser999/Public-ID-Deserialisation#public-id-deserialisation) earlier to test out what options we have.  My preferred choice would be option 2 from there, but I haven't checked to see if that works with serde v0.9 yet.  I was hoping to get a chance to discuss this with the team soon.